source/errors.ts

```diff
@@ -22,36 +49,34 @@ class GotError extends Error {
 			socketPath: options.socketPath,
 			protocol: options.protocol,
 			url: options.href,
-			gotOptions: options
+			gotOptions: options.options
 		});
 	}
 }
 
-module.exports.GotError = GotError;
-
-module.exports.CacheError = class extends GotError {
-	constructor(error, options) {
+export const CacheError = class extends GotError {
+	constructor(error: any, options: Options) {
 		super(error.message, error, options);
 		this.name = 'CacheError';
 	}
 };
 
-module.exports.RequestError = class extends GotError {
-	constructor(error, options) {
+export const RequestError = class extends GotError {
+	constructor(error: any, options: Options) {
 		super(error.message, error, options);
 		this.name = 'RequestError';
 	}
 };
 
-module.exports.ReadError = class extends GotError {
-	constructor(error, options) {
+exports.ReadError = class extends GotError {
+	constructor(error: any, options: Options) {
 		super(error.message, error, options);
 		this.name = 'ReadError';
 	}
 };
 
-module.exports.ParseError = class extends GotError {
-	constructor(error, statusCode, options, data) {
+exports.ParseError = class extends GotError {
+	constructor(error: any, statusCode: number, options: Options, data: any) {
```

Comment:
We need a better type for `data`. Can you console log it and check what possible types it might receive and then use that?

Reply:
Hi @sindresorhus, I need help here, I've been trying to understand how to console log it.
I created a node js example using the Got library wiht a `post` request. I readed about **ParseError** in docs

> When server response code is 2xx, and parsing body fails. Includes body, statusCode and statusMessage properties.

Could you tell me more detail about it. 

Thank you for your patience :confused:

Reply:
@sindresorhus I added a console.log and ran `npm test`, nothing is logged.

```typescript
export class ParseError extends GotError {
	statusCode: number;
	constructor(error: any, statusCode: number, options: any, data: any) {
                console.log('=======', typeof data);
		super(`${error.message} in "${urlLib.format(options)}"`, error, options);
		this.name = 'ParseError';
		this.body = data;
		this.statusCode = statusCode;
		this.statusMessage = http.STATUS_CODES[this.statusCode];
	}
}
```

Could you provide more thoughts :confused: